src/modules/commander/Commander.cpp

```diff
@@ -3322,17 +3324,27 @@ set_control_mode()
 		 * The control flags depend on what is ignored according to the offboard control mode topic
 		 * Inner loop flags (e.g. attitude) also depend on outer loop ignore flags (e.g. position)
 		 */
-		control_mode.flag_control_rates_enabled = !offboard_control_mode.ignore_bodyrate ||
-				!offboard_control_mode.ignore_attitude ||
-				!offboard_control_mode.ignore_position ||
-				!offboard_control_mode.ignore_velocity ||
-				!offboard_control_mode.ignore_acceleration_force;
+		control_mode.flag_control_rates_enabled =
+			!offboard_control_mode.ignore_bodyrate_x ||
+			!offboard_control_mode.ignore_bodyrate_y ||
+			!offboard_control_mode.ignore_bodyrate_z ||
+			!offboard_control_mode.ignore_attitude ||
+			!offboard_control_mode.ignore_position ||
+			!offboard_control_mode.ignore_velocity ||
+			!offboard_control_mode.ignore_acceleration_force;
 
 		control_mode.flag_control_attitude_enabled = !offboard_control_mode.ignore_attitude ||
 				!offboard_control_mode.ignore_position ||
 				!offboard_control_mode.ignore_velocity ||
 				!offboard_control_mode.ignore_acceleration_force;
 
+		// TO-DO: Add support for other modes than yawrate control
```

Comment:
I'm not sure about this todo. From my tests other rate control already works.

Reply:
We are controlling in yawrate, but I am not sure if it would be of interest to have pitchrate and roll rate, combined with other attitude angles. Anyhow, I don't think it is, so maybe we can remove the to-do.

Reply:
I think we can remove it because when I test with the Dronecode SDK it roll and pitch rate control works correctly.